dsnp-wallet/dsnp-wallet/Source/ViewControllers/ProfileViewController.swift

```diff
@@ -5,10 +5,366 @@
 //  Created by Ryan Sheh on 4/26/22.
 //
 
-import Foundation
 import UIKit
 
 class ProfileViewController: UIViewController {
+    //MARK: Constants
+    private let leadingConstraintConstant = 20
+    private let saveBtnBottomOffset = 20
+    private let profileText = "Profile"
+    private let bioText = "Bio"
     
+    //MARK: UI
+    private var profileHeaderView: ProfileHeaderView?
+    private var emailTextfield: UITextField?
+    private var bioTextfield: UITextField?
+    private let saveBtn = SharedButton()
+    private var saveViewBottomConstraint: NSLayoutConstraint!
+    private var scrollViewBottomConstraint: NSLayoutConstraint!
+    
+    //MARK: Blocks
+    var didEmailChange: Bool = false {
+        didSet {
+            self.saveBtn.enable()
+        }
+    }
+    var didTogglesChange: Bool = false {
+        didSet {
+            self.saveBtn.enable()
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setViews()
+    }
+}
+
+//MARK: Outlets
+extension ProfileViewController {
+    @objc func saveChanges(selector: UIButton?) {
+        
+    }
+}
+
+//MARK: UI Helper Funcs
+extension ProfileViewController {
+    private func setViews() {
+        self.view.backgroundColor = UIColor.Theme.background
+        
+        let stackView = getScrollableStackView()
+        
+        let profileHeaderView = ProfileHeaderView(parent: self)
+        profileHeaderView.delegate = self
+        self.profileHeaderView = profileHeaderView
+    
+        let profileUsernameView = getTextfieldViewWith(titleLabel: profileText, placeholder: "Add Profile Name")
+        let bioView = getTextfieldViewWith(titleLabel: bioText, placeholder: "Add Bio")
+        let permissionsView = getPermissionsView()
+        
+        stackView.addArrangedSubview(profileHeaderView)
+        stackView.addArrangedSubview(SharedSpacer(height: 6))
+        stackView.addArrangedSubview(profileUsernameView)
+        stackView.addArrangedSubview(bioView)
+        stackView.addArrangedSubview(SharedSpacer(height: 3))
+        stackView.addArrangedSubview(permissionsView)
+        
+        setSaveBtn()
+        setKeyboard()
+    }
+    
+    private func getScrollableStackView() -> UIStackView {
+        let scrollView = UIScrollView()
+        view.addSubview(scrollView)
+        scrollView.translatesAutoresizingMaskIntoConstraints = false
+        scrollView.leadingAnchor.constraint(equalTo: view.leadingAnchor).isActive = true
+        scrollView.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        scrollView.topAnchor.constraint(equalTo: view.layoutMarginsGuide.topAnchor).isActive = true
+        scrollViewBottomConstraint = NSLayoutConstraint(item: scrollView,
```

Comment:
Mixing constraint methods. Can we replace with scrollView.bottomAnchor.constraint(..) ?

Reply:
I used a nslayoutconstraint property because its updated with the keyboard. I previously tried attaching the constraint to view.keyboardlayoutguide.topanchor, but that didn't seem to work well when switching tabs for some reason.